src/plugins/blocksmith/components/instagram-media-block-preview/index.jsx

```diff
@@ -1,17 +1,84 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Message from '@gdbots/pbj/Message';
-import InstagramEmbed from 'react-instagram-embed';
+import './styles.scss';
 
-const InstagramMediaBlockPreview = ({ block }) => (
-  <InstagramEmbed
-    url={`https://instagr.am/p/${block.get('id')}/`}
-    hideCaption={block.get('hidecaption') || false}
-  />
-);
+class InstagramPostBlockPreview extends Component {
+  constructor(props) {
+    super(props);
+    this.embedParentRef = React.createRef();
+    this.clean = this.clean.bind(this);
+    this.embed = this.embed.bind(this);
+  }
 
-InstagramMediaBlockPreview.propTypes = {
+  componentDidMount() {
+    this.addInstagramEmbedScript();
+    this.embed();
+  }
+
+  componentDidUpdate() {
+    this.embed();
+    if (window.instgrm) {
+      window.instgrm.Embeds.process();
+    }
+  }
+
+  componentWillUnmount() {
+    this.clean();
+  }
+
+  addInstagramEmbedScript() {
+    const script = document.createElement('script');
+    script.async = true;
+    script.src = '//www.instagram.com/embed.js';
+    script.setAttribute('charset', 'utf-8');
+    script.onload = () => {
+      window.instgrm.Embeds.process();
+    };
+    this.embedParentRef.current.appendChild(script);
+  }
+
+  clean() {
+    const { current } = this.embedParentRef;
+    Array.from(current.children).forEach((child) => {
+      if (child.className.indexOf('instagram-embed') === -1) {
```

Comment:
@tbone1020 this is unnecessary, it is actually ok to remove the script as well because window.instgrm has already been added to the site, so we can remove it here and not worry about removing it later